web/src/reload.ts

```diff
@@ -58,9 +58,8 @@ function preserve_state(send_after_reload: boolean, save_compose: boolean): void
         const msg_type = compose_state.get_message_type();
         if (msg_type === "stream") {
             const stream_id = compose_state.stream_id();
-            assert(stream_id !== undefined);
             url += "+msg_type=stream";
-            url += "+stream_id=" + encodeURIComponent(stream_id);
+            url += "+stream_id=" + (stream_id ? encodeURIComponent(stream_id) : undefined);
```

Comment:
Does a string ID of undefined work in the code that reads this preserved state?

Reply:
I guess the commit message says you tested this, so merging.

Reply:
It works (checked manually), but I didn't check (in the code) that it works for reasons that seem reasonable. Mostly I was switching this back to the unbroken state it was in before the typescript migration. I guess it would be helpful to see what's happening on the other side.

```js
            compose_actions.start({
                message_type: vars.msg_type,
                stream_id: Number.parseInt(vars.stream_id, 10) || undefined,
                topic: vars.topic || "",
                private_message_recipient: vars.recipient || "",
                content: vars.msg || "",
                draft_id: vars.draft_id || "",
            });
```

Yeah probably we just shouldn't send the stream id at all.

Reply:
#31794